src-newer-jdks/orchard/java/parser.clj

```diff
@@ -230,14 +230,14 @@
         (when-let [^DocletEnvironment root (parse-java path (module-name klass))]
           (try
             (let [path-resource (io/resource path)]
-              (assoc (->> (.getIncludedElements root)
-                          (filter #(#{ElementKind/CLASS
-                                      ElementKind/INTERFACE
-                                      ElementKind/ENUM}
-                                    (.getKind ^Element %)))
-                          (keep #(parse-info % root))
-                          (filter #(= klass (:class %)))
-                          (first))
```

Comment:
This is a bit better due to laziness, but it will still realize at least a 32-chunk of `parse-info`s before doing `first` on them.